core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -132,11 +122,28 @@ public RegistryService getRegistryService() {
      */
     public static KeyStoreManager getInstance(int tenantId) {
 
-        log.debug("KeyStoreManager Instace created for teant id: " + tenantId);
-        return getInstance(tenantId, CarbonCoreDataHolder.getInstance().
-                getServerConfigurationService(), CryptoUtil.lookupRegistryService());
+        log.debug("KeyStoreManager Instance created for tenant id: " + tenantId);
+        CarbonUtils.checkSecurity();
+        String tenantIdString = Integer.toString(tenantId);
+        if (!mtKeyStoreManagers.containsKey(tenantIdString)) {
+            mtKeyStoreManagers.put(tenantIdString, new KeyStoreManager(tenantId, CarbonCoreDataHolder.getInstance().
+                    getServerConfigurationService(), CryptoUtil.lookupRegistryService()));
+        }
+        return mtKeyStoreManagers.get(tenantIdString);
```

Comment:
Lets extract this to a method to avoid code duplication.

Reply:
Addressed with https://github.com/wso2/carbon-kernel/compare/63f3d33d1ed8704022f55645be6b047a005641fb..6c4a7e5fc5efca69a114e5d3a08289b8dc809be7